persistence/sql/persister_consent.go

```diff
@@ -364,74 +401,91 @@ func (p *Persister) GetRememberedLoginSession(ctx context.Context, id string) (*
 	return &s, nil
 }
 
-func (p *Persister) ConfirmLoginSession(ctx context.Context, id string, authenticatedAt time.Time, subject string, remember bool) error {
+func (p *Persister) ConfirmLoginSession(ctx context.Context, session *flow.LoginSession, id string, authenticatedAt time.Time, subject string, remember bool) error {
 	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.ConfirmLoginSession")
 	defer span.End()
 
-	_, err := p.Connection(ctx).Where("id = ? AND nid = ?", id, p.NetworkID(ctx)).UpdateQuery(&consent.LoginSession{
+	// If we previously cached the login session, we now want to persist it to db.
+	if session != nil {
```

Comment:
Not quite sure if I follow here - in which case would this be false?

As far as I understand, `session` is coming from the new cookie, and would always be defined?

Reply:
We are still using that path in some tests.

Reply:
Can you please add a small comment? TY